src/utils/collection/apply.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const arraySize = require('../array').size
+const isMatrix = require('../collection/isMatrix')
+const IndexError = require('../../error/IndexError')
+
+/**
+ * Apply function that takes an array and returns a Scalar
+ * along a given axis of a matrix or array.
+ * Returns a new matrix or array with one less dimension.
+ * @param {Array | Matrix} mat
+ * @param {number} dim
+ * @param {Function} callback
+ * @return {Array | Matrix} res
```

Comment:
Would it be possible to give fuller descriptions of these parameters?

Reply:
Yes, will do.